### PR TITLE
fix: implies should not fail when implied key's value is 0, false or empty string

### PR DESCRIPTION
**Pull request by mathieubergeron**
This is an attemps at fixing #1091 and #1203.

## The issue

The documentation for `implies` is:

> Given the key x is set, it is required that the key y is set.

But this check fails when the value of the key `y` is `0`, `false`, or empty string.

When `y` value is `false`, we might want `implies` to continue failing though. I'd like to have your input on that, and I will adjust that PR accordingly.

The issue came from `keyExists` which, instead of always returning a boolean, could returned the value of the key. That value is later on used like so:
```
if (key && !value) {
  implyFail.push(` ${origKey} -> ${origValue}`);
}
```
Thus, that `if` resolved to `true` when `value` was `0`, `false` or `""`.

## The fix

Having `keyExists` always returning a boolean seems more robust. I used `Object.hasOwnProperty` to check for presence of the key instead.

Let me know if you think that `keyExists` should still return `false` when the value of the key is `false`.

Thanks :)

### CODE REVIEW

**Comment by bcoe** [comment]
@mathieubergeron thanks for the contribution 👍 will make an effort to give review soon.

**Comment by bcoe** [comment]
@jly36963 if you happen to beat me to code review, at a glance this contribution looked solid to me 😄

File: test/validation.cjs [hunk 1]
```diff
@@ -101,6 +101,28 @@ describe('validation tests', () => {
       failCalled.should.equal(true);
     });
 
+    it("doesn't fail if implied key exists with value 0", () => {
+      let failCalled = false;
+      yargs('--foo --bar 0')
+        .implies('foo', 'bar')
+        .fail(msg => {
+          failCalled = true;
```
Comment (jly36963) [inline test/validation.cjs:109]: You could probably remove `failCalled` declaration/update/assertion and instead use `chai.expect.fail()` in the fail callback. `expect` should already be imported from chai.

```js
.fail(() => {
  expect.fail()
}
```

**Comment by jly36963** [comment]
@bcoe I tested it locally with and without his changes, his fix works.

**Review by jly36963 — APPROVED** [review]

**Review by bcoe — APPROVED** [review]

**Pull request merged**